### PR TITLE
Facebook login

### PR DESCRIPTION
**Pull request by DKbyo**
- App page https://developers.facebook.com/apps/651280398386350
- Login auto sign up a user if user doesn't exit, this sign up only add the user with the field email and user_id.
- Validates the facebook userid and access_token in the server, so the server needs internet connection.

### CODE REVIEW

File: app/ds/user.go [hunk 1]
```diff
@@ -90,6 +93,75 @@ func Delete(userID uint) (err error) {
...
+		err = errors.New(`{"facebook":"token-invalid"}`);
+	}
+	if err!=nil {
+		return
+	}
+	err = userSource.Find("email", logRequest.FacebookID).One(&user)
+	if err != nil {
+		err = nil;
+		user.Email =  logRequest.FacebookID;
+		user.FirstName = "";
```
Comment (manue1) [inline app/ds/user.go]: Can we set the first and last name fields?

Comment (manue1) [inline app/ds/user.go]: Can we set the first and last name fields?

File: app/ds/user.go [hunk 2]
```diff
@@ -90,6 +93,75 @@ func Delete(userID uint) (err error) {
...
+		user.LastName = "";
+		user.Answers = logRequest.Answers;
```
Comment (manue1) [inline app/ds/user.go:134]: The answers currently don't get sent to the User API on login, however I think we should do this and also add it to the general login.

However the 'result' values within the answers body need to be separately saved in the total_footprint field. (see [UpdateAnswers()](https://github.com/arbolista-dev/cc-user-api/blob/master/app/controllers/user.go#L171))

File: app/ds/user.go (hunk 1 above)
Comment (DKbyo) [reply app/ds/user.go]: Done!

**Pull request merged**